config/yaml_test.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestMetricDefValidate(t *testing.T) {
+	one := 1
+	for _, test := range []struct {
+		name        string
+		metricDef   MetricDef
+		expectedErr error
+	}{
+		{
+			"bool",
+			MetricDef{
+				DataType: ModbusBool,
+			},
+			nil,
+		},
+		{
+			"bool",
+			MetricDef{
+				DataType:    ModbusInt16,
+				BitPosition: &one,
+			},
+			fmt.Errorf("bitPosition can only be used with boolean data type"),
+		},
+	} {
+		err := test.metricDef.validate()
+
+		if err != test.expectedErr {
+			if err == nil || test.expectedErr == nil {
+				t.Fatalf("expected err to be %v but got %v", test.expectedErr, err)
+			}
+
+			if err.Error() != test.expectedErr.Error() {
+				t.Fatalf("expected err to be %v but got %v", test.expectedErr, err)
+			}
+		}
+	}
+}
+
+func TestMetricDefParse(t *testing.T) {
+	one := 1
+	two := 2
+
+	tests := []struct {
+		name          string
+		input         func() [2]byte
+		metricDef     func() *MetricDef
+		expectedValue float64
+	}{
+		{
+			name: "bool, no bit",
```

Comment:
@RichiH here you see couple of examples of the current _boolean data type_ implementation.

Reply:
Yes, this looks sane. `input[1] = uint8(1)` should then have `1` at `BitPosition: 9` aka zero-indexed 8.